src/main/java/io/kestra/plugin/kestra/executions/Search.java

```diff
@@ -0,0 +1,225 @@
+package io.kestra.plugin.kestra.executions;
+
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.property.Property;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.models.tasks.common.FetchOutput;
+import io.kestra.core.models.tasks.common.FetchType;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.serializers.FileSerde;
+import io.kestra.plugin.kestra.AbstractKestraTask;
+import io.kestra.sdk.KestraClient;
+import io.kestra.sdk.internal.ApiException;
+import io.kestra.sdk.model.*;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.annotation.Nullable;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+import reactor.core.publisher.Flux;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+@SuperBuilder(toBuilder = true)
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Search for Kestra Executions"
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Search for execution with specific label>",
+            full = true,
+            code = """
+                id: search_executions_by_label
+                namespace: company.team.myflow
+                
+                tasks:
+                  - id: hello
+                    type: io.kestra.plugin.kestra.executions.Search
```

Comment:
```suggestion
                    type: io.kestra.plugin.kestra.executions.Query
```
While I'm aware we have a search endpoint, the goal of the task is to look-up, query, or list executions. Search somehow imply looking for a specific executions and we're looking for a list/query to return all results matching query filters. 

Strongly recommend picking one of these:
1. Lookup
2. Query
3. List

Since we use fetchType, `Query` seems to be a reasonable choice.